### PR TITLE
Adding the Idobao ID96 keyboard

### PR DESCRIPTION
**Pull request by vinorodrigues**
## Description

Adding the Idobao ID96 keyboard.  This keyboard uses the same PCB as the YMDK Melody 96 keyboard, but I felt it prudent to add specific firmware for the ID96, with a few minor of changes:

* Keyboard USB identifier is now aligned with the other Idobao keyboards - `VENDOR_ID 0x6964 // "id"`, etc.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard *(addition or update)*
- [x] Keymap/layout/userspace *(addition or update)*
- [ ] Documentation

## Issues Fixed or Closed by This PR

n/a 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything *(as well as I can manage)*.

### CODE REVIEW

File: keyboards/idobao/id96/id96_via_sideload_kle.json [hunk 1]
```diff
@@ -0,0 +1,623 @@
+{
```
Comment (zvecr) [inline keyboards/idobao/id96/id96_via_sideload_kle.json:1]: Needs to be removed, we do not accept VIA config files to this repo.

File: keyboards/idobao/id96/readme.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## This build also supports [VIA](https://caniusevia.com/)...
+
+...but VIA does not support it.
+
+VIA will not detect the keyboard in VIA's default state - you will need the VIA KLE JSON file for *"sideloading"* the keyboard into VIA.
+
+*(You will need to load this file each time you start VIA.)*
+
+In VIA, use *"Import Keymap"* and load the file: [`id96_via_sideload_kle.json`](id96_via_sideload_kle.json).
```
Comment (zvecr) [inline keyboards/idobao/id96/readme.md:40]:
```suggestion
# Idobao ID96

![Idobao ID96](https://cdn.shopify.com/s/files/1/0382/1348/4675/products/96.png?v=1620272768)

96-key Keyboard from Idobao. The PCB for this board is also commonly used in the Drop NYM96 and YMDK Melody96

* Keyboard Maintainer: [Vino Rodrigues](https://github.com/vinorodrigues)
* Hardware Supported: Idobao ID96
* Hardware Availability: [Idobao ID96](https://www.idobao.net/collections/90/products/idobao-id96-aluminum-keyboard-kit)

Make example for this keyboard (after setting up your build environment):

    make idobao/id96:default

Flashing example for this keyboard:

    make idobao/id96:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/idobao/id96/rules.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+BOOTMAGIC_ENABLE = lite  # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no     # Mouse keys
+EXTRAKEY_ENABLE = yes    # Audio control and System control
+CONSOLE_ENABLE = no      # Console for debug
+COMMAND_ENABLE = no      # Commands for debug and configuration
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes        # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes   # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no     # Althoug the board has RGB bottom lights, these are covered by the case bottom.
+AUDIO_ENABLE = no        # No sound
```
Comment (zvecr) [inline keyboards/idobao/id96/rules.mk:26]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/idobao/id96/config.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+// #define RGB_DI_PIN E2  // RGB backlight LEDs are covered by the backplate, so can't be seen
```
Comment (zvecr) [inline keyboards/idobao/id96/config.h:66]: We care about PCB, not case, so if a keyboard has RGB it should be enabled by default
```suggestion
#define RGB_DI_PIN E2
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/idobao/id96/id96_via_sideload_kle.json (hunk 1 above)
Comment (vinorodrigues) [reply keyboards/idobao/id96/id96_via_sideload_kle.json]: deleted with commit 8e4ec92

File: keyboards/idobao/id96/config.h [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#define RGB_DI_PIN E2
+#ifdef RGB_DI_PIN
+  #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/idobao/id96/config.h:68]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/idobao/id96/config.h (hunk 5 above)
Comment (vinorodrigues) [reply keyboards/idobao/id96/config.h]: done

**Review by drashna — APPROVED** [review]

File: keyboards/idobao/id96/rules.mk [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/idobao/id96/rules.mk:21]:
```suggestion
```

File: keyboards/idobao/id96/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+// VIA supports 4 layers
+
```
Comment (fauxpark) [inline keyboards/idobao/id96/keymaps/via/keymap.c:22]:
```suggestion
```

File: keyboards/idobao/id96/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        _______, RGB_TOG, _______, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, _______, _______, _______, _______,          _______, _______, _______, _______,
+        BL_TOGG, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______,
+        _______, _______, _______, _______, BL_DEC,  BL_TOGG, BL_INC,  _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______, _______, _______, _______, _______),
+};
+
+
+// bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+//     return true;
+// }
```
Comment (fauxpark) [inline keyboards/idobao/id96/keymaps/default/keymap.c:95]:
```suggestion
```

File: keyboards/idobao/id96/info.json [hunk 9]
```diff
@@ -0,0 +1,121 @@
+{
+    "keyboard_name": "id96",
+    "url": "https://www.idobao.net/collections/90/products/idobao-id96-aluminum-keyboard-kit",
+    "maintainer": "vinorodrigues",
+    "width": 19,
+    "height": 6,
```
Comment (fauxpark) [inline keyboards/idobao/id96/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by vinorodrigues** [comment]
:) you're welcome.

**Pull request merged**